hamilflow/models/brownian_motion.py

```diff
@@ -183,17 +185,29 @@ def _trajectory(self, n_new_steps: int, seed: int) -> np.ndarray:
 
         return trajectory
 
-    def __call__(self, n_steps: int, seed: int = 42) -> pd.DataFrame:
-        """Simulate the coordinates of the particle
+    def generate_from(self, n_steps: int, seed: int = 42) -> pd.DataFrame:
+        """generate data from a set of interpretable params for this model
 
         :param n_steps: total number of steps to be simulated, including the inital step.
         :param seed: random generator seed for the stochastic process.
             Use it to reproduce results.
         """
+        time_steps = np.arange(0, n_steps) * self.system.delta_t
+
+        return self(t=time_steps, seed=seed)
+
+    def __call__(self, t: TypeTime, seed: int = 42) -> pd.DataFrame:
+        """Simulate the coordinates of the particle
+
+        :param t: the time sequence to be used to generate data
+        :param seed: random generator seed for the stochastic process.
+            Use it to reproduce results.
+        """
+        n_steps = len(t)
```

Comment:
This definitely crashes for scalar `t`. Please add a unit test and fix it.

Reply:
fixed in f70b6ac